File: Selenium/Project/AddQualification.java
package HRM;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class AddQualification {
	
	static WebDriver driver;
	static Actions action;
	static WebDriverWait wait;
	
	@BeforeClass
	public void setUp() {
		driver = new ChromeDriver();
		driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");
		driver.manage().window().maximize();
		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
		wait = new WebDriverWait(driver, Duration.ofSeconds(5));
	}
	
	@Test(priority =1)
	@Parameters({"username", "password"})
	public void login_page(String username, String password) {
		driver.findElement(By.id("txtUsername")).sendKeys(username);
		driver.findElement(By.id("txtPassword")).sendKeys(password);
		driver.findElement(By.id("btnLogin")).click();
		Boolean logged_in = driver.findElement(By.xpath("//*[@id = 'welcome']")).isDisplayed();
		Assert.assertTrue(logged_in);
	}
	
	@Test(priority =2)
	public void add_qualifications() throws InterruptedException {
		
		wait.until(ExpectedConditions.elementToBeClickable(By.id("menu_pim_viewMyDetails")));
		driver.findElement(By.id("menu_pim_viewMyDetails")).click();
		
		wait.until(ExpectedConditions.elementToBeClickable(By.linkText("Qualifications")));
		driver.findElement(By.linkText("Qualifications")).click();
		
		wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//h1[text() = 'Work Experience']")));
		driver.findElement(By.id("addWorkExperience")).click();
		driver.findElement(By.id("experience_employer")).sendKeys("abc Company");
		Thread.sleep(10000);
		driver.findElement(By.id("experience_jobtitle")).sendKeys("Tester");
		WebElement fromDate = driver.findElement(By.id("experience_from_date"));
		fromDate.clear();
		fromDate.sendKeys("2021-03-01");
		WebElement toDate = driver.findElement(By.id("experience_to_date"));
		toDate.clear();
		toDate.sendKeys("2024-01-01");
		driver.findElement(By.id("experience_comments")).sendKeys("Tester Comments");
		driver.findElement(By.id("btnWorkExpSave")).click();
		
	}
	
	@AfterClass
	public void tearDown() {
		
		driver.quit();
	}

}

File: Selenium/Project/AddQualification.py
import time

from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait

def add_qualifications(driver, wait):
    wait.until(expected_conditions.element_to_be_clickable((By.ID, "menu_pim_viewMyDetails")))
    driver.find_element(By.ID, "menu_pim_viewMyDetails").click()
    wait.until(expected_conditions.element_to_be_clickable((By.LINK_TEXT, "Qualifications")))
    driver.find_element(By.LINK_TEXT, "Qualifications").click()
    wait.until(expected_conditions.visibility_of_element_located((By.XPATH, "//h1[text() = 'Work Experience']")))
    driver.find_element(By.ID, "addWorkExperience").click()
    driver.find_element(By.ID, "experience_employer").send_keys("def company")
    driver.find_element(By.ID, "experience_jobtitle").send_keys("Testing")
    from_date = driver.find_element(By.ID, "experience_from_date")
    from_date.clear()
    from_date.send_keys("2021-08-04")
    to_date = driver.find_element(By.ID, "experience_to_date")
    to_date.clear()
    to_date.send_keys("2021-08-04")
    driver.find_element(By.ID, "experience_comments").send_keys("Testing Comments")
    driver.find_element(By.ID, "btnWorkExpSave").click()

    time.sleep(5)

with webdriver.Chrome() as driver:
    driver.get("http://alchemy.hguy.co/orangehrm")
    title = driver.title
    print(title)

    driver.find_element(By.ID, "txtUsername").send_keys("orange")
    driver.find_element(By.ID, "txtPassword").send_keys("orangepassword123")
    driver.find_element(By.ID, "btnLogin").click()
    assert driver.find_element(By.XPATH, "//*[@id = 'welcome']").is_displayed();
    print(driver.find_element(By.TAG_NAME, "h1").text)
    wait = WebDriverWait(driver, 5)
    action = ActionChains(driver)
    add_qualifications(driver, wait)

    driver.close()
